### PR TITLE
feat: Enable macos desktop

### PR DESCRIPTION
**Pull request by ymauray**
# Description

This pull request enables macos desktop for `quickgui`.

**This is still early work, so they're might be some edges cases that will need investigation.**

## Type of change

- [X] New feature (non-breaking change which adds functionality)

# Checklist:

- [X] I have performed a self-review of my code
- [X] I have tested my code in common scenarios and confirmed there are no regressions
- [X] I have added comments to my code, particularly in hard-to-understand sections
- [X] I have updated and committed `pubspec.yaml` and `pubspec.lock`
  - `flutter pub get`

# What is has been tested on 
- mac book air M2, default macos terminal

# What works
- Getting the list of available OSes and their versions from `quickget`
- Starting the download, and monitoring its progression
- Listing availables VMs
- Starting and stopping a VM through `quickemu`
- Connecting to a VM via SSH using the default macos terminal
- Partially or completely deleting a VM

# What has not been tested
- Downloading and running any distro which does not have "ubuntu" in its name

# What does not work
- Packaging a dmg file. I need to look into that.

### CODE REVIEW

File: macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_1024.png
Comment (flexiondotorg) [inline macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_1024.png:1]: Should this icon be a Quickemu/gui logo?

File: macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_128.png
Comment (flexiondotorg) [inline macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_128.png:1]: Should this icon be a Quickemu/gui logo?

File: macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_16.png
Comment (flexiondotorg) [inline macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_16.png:1]: Should this icon be a Quickemu/gui logo?

File: macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_256.png
Comment (flexiondotorg) [inline macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_256.png:1]: Should this icon be a Quickemu/gui logo?

File: macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_32.png
Comment (flexiondotorg) [inline macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_32.png:1]: Should this icon be a Quickemu/gui logo?

File: macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_512.png
Comment (flexiondotorg) [inline macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_512.png:1]: Should this icon be a Quickemu/gui logo?

File: macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_64.png
Comment (flexiondotorg) [inline macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_64.png:1]: Should this icon be a Quickemu/gui logo?

**Review by flexiondotorg — COMMENTED** [review]
I'll find time to build/test this soon :+1:

File: macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_1024.png
Comment (ymauray) [reply macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_1024.png:1]: yes, yes it should !

File: macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_128.png
Comment (ymauray) [reply macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_128.png:1]: yes, yes it should !

File: macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_16.png
Comment (ymauray) [reply macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_16.png:1]: yes, yes it should !

File: macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_256.png
Comment (ymauray) [reply macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_256.png:1]: yes, yes it should !

File: macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_32.png
Comment (ymauray) [reply macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_32.png:1]: yes, yes it should !

File: macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_512.png
Comment (ymauray) [reply macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_512.png:1]: yes, yes it should !

File: macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_64.png
Comment (ymauray) [reply macos/Runner/Assets.xcassets/AppIcon.appiconset/app_icon_64.png:1]: yes, yes it should !

**Comment by ymauray** [comment]
I fixed the icons, hopefully.

**Review by ymauray — COMMENTED** [review]
This seems to do the trick for building the .dmg

**Review by flexiondotorg — APPROVED** [review]

**Comment by gregpalaci** [comment]
Sorry to reopen I'm 99% way for compling on arm64 mac but getting a weird error.
I build with Dart and quickemu is in the Path.... any tips?

Screenshot 2024-09-17 at 7 42 12 PM
Screenshot 2024-09-17 at 8 02 53 PM

**Pull request merged**